integration-tests/oidc-wiremock/src/test/java/io/quarkus/it/keycloak/BearerTokenAuthorizationTest.java

```diff
@@ -0,0 +1,122 @@
+package io.quarkus.it.keycloak;
+
+import static org.hamcrest.Matchers.equalTo;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+import io.smallrye.jwt.build.Jwt;
+
+/**
+ * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
+ */
+@QuarkusTest
+@QuarkusTestResource(KeycloakTestResource.class)
+public class BearerTokenAuthorizationTest {
+
+    @Test
+    public void testSecureAccessSuccessWithCors() {
+        String origin = "http://custom.origin.quarkus";
+        String methods = "GET";
+        String headers = "X-Custom";
+        RestAssured.given().header("Origin", origin)
+                .header("Access-Control-Request-Method", methods)
+                .header("Access-Control-Request-Headers", headers)
+                .when()
+                .options("/api").then()
+                .statusCode(200)
+                .header("Access-Control-Allow-Origin", origin)
+                .header("Access-Control-Allow-Methods", methods)
+                .header("Access-Control-Allow-Headers", headers);
+
+        for (String username : Arrays.asList("alice", "jdoe", "admin")) {
+            RestAssured.given().auth().oauth2(getAccessToken(username, new HashSet<>(Arrays.asList("user", "admin"))))
+                    .when().get("/api/users/preferredUserName")
+                    .then()
+                    .statusCode(200)
+                    .body("userName", equalTo(username));
+        }
+    }
+
+    //    @Test
+    //    public void testSecureAccessSuccessCustomPrincipal() {
```

Comment:
To get this passing you just need to set the `email` claim with the user name which is what the `application.properties`, but it would be simpler to remove this test as it is not really stressing the mock more than other tests